src/usdb_syncer/song_loader.py

```diff
@@ -339,3 +343,88 @@ def _maybe_write_txt(ctx: Context) -> None:
 
 def _write_sync_meta(ctx: Context) -> None:
     ctx.sync_meta.to_file(ctx.locations.dir_path())
+
+
+def _maybe_write_audio_tags(ctx: Context) -> None:
+    if not (options := ctx.options.audio_options) or not (meta := ctx.sync_meta.audio):
+        return
+
+    audiofile = ctx.locations.file_path(meta.fname)
+
+    match audiofile.suffix:
+        case ".m4a":
+            _write_m4a_tags(meta, ctx, options.embed_artwork)
+        case ".mp3":
+            _write_mp3_tags(meta, ctx, options.embed_artwork)
+
+
+def _write_m4a_tags(audio_meta: FileMeta, ctx: Context, embed_artwork: bool) -> None:
+    path = ctx.locations.file_path(audio_meta.fname)
+    m4a = mutagen.mp4.MP4(path)
```

Comment:
Shouldn't `path` be removed as in `_write_mp3_tags()`? Below you check `if m4a.tags`. Does this mean no tags will be written if the file does not already have tags?

Reply:
It seems that `mutagen.mp4.MP4()` needs an actual file as parameter (there is no equivalent to ID3() for m4a files) and `m4a = mutagen.mp4.MP4()` does not work, while `tags = id3.ID3()` creates an empty ID3 object, then fills it and then writes it to the file. Not sure if `if m4a.tags` is needed, but if I leave it out, I get `Unsupported target for indexed assignment ("None")  [index]mypy(error)` errors.

Reply:
Ok, so the API requires it for whatever reason.
As for the second point, why not create and assign a new `MP4Tags` instance?

Reply:
Good point. I can try if that works as well.

Reply:
Seems to work. Just pushed the changes.